src/FwdState.cc

```diff
@@ -762,10 +762,16 @@ FwdState::doneWithRetries()
 }
 
 // called by the server that failed after calling unregister()
+// err may or may not be null here depending on the type of server end.
 void
 FwdState::handleUnregisteredServerEnd()
 {
-    debugs(17, 2, HERE << "self=" << self << " err=" << err << ' ' << entry->url());
+    if(err) {
+        const auto errType = errorTypeName(err->type);
+        debugs(17, 2, HERE << "self=" << self << " err=" << errType << ' ' << entry->url());
+    } else {
+        debugs(17, 2, HERE << "self=" << self << " err=" << err << ' ' << entry->url());
+    }
```

Comment:
Please undo this entire FwdState change because it adds too much always-executed code and noise for debugging sake.

If you really want to improve `err` debugging, overload the corresponding `<<` operator instead (in errorpage.{h,cc} files), keeping the above debugs() line intact. There is a somewhat similar example already (see the first bullet below), but this new _public_ operator would have to print something like `[none]` when there is no error (see the second bullet for inspiration). The operator will improve debugging in several other places (see the third bullet for an incomplete dirty list of candidates -- you can and, ideally, should find (and, if needed, polish) all of them by [deleting](https://godbolt.org/z/4aaWhTK4G) the printing operator first).

* `git grep 'operator <<.*[*]'`
* `git grep '<< "[[]n'`
* `git grep '<< err\b'`

I insist on the removal, but I do not insist on adding the operator. I can volunteer to add it, but it will take some time.

Reply:
As I am unfamiliar with how operator overloading works in C++, I believe it is best for me to just remove this, as I suspect any changes made by my in this case will just be a (frustrating) game of cat-and-mouse.